src/clj/web/user.clj

```diff
@@ -9,6 +9,28 @@
    :isadmin :ismoderator :tournament-organizer
    :special :options :stats :has-api-keys :banned])
 
+(def max-username-chars 20)
+(def ascii-invalid-char-pattern #"[^\w.-]")
+(def unicode-invalid-char-pattern #"[^\p{Alnum}\p{M}._-]")
+(def invalid-leading-char-pattern #"^[^\p{Alnum}\p{M}_]+")
+(def invalid-trailing-char-pattern #"[^\p{Alnum}\p{M}]+$")
+(def repeated-special-char-pattern #"[-_.]{2,}")
+(def patterns-username [
```

Comment:
I'd prefer this to be on the next line and the white space trimmed so it's:
```clojure
(def patterns
  [pattern-1
   ...
   pattern-X])
```

Reply:
Yeah, I just did that because my editor auto-did it, and figured it knows better than me for Clojure code :laughing: